Framework/Algorithms/src/ConvertToConstantL2.cpp

```diff
@@ -104,7 +114,9 @@ void ConvertToConstantL2::exec() {
     oldPos.getSpherical(r, theta, phi);
     V3D newPos;
     newPos.spherical(m_l2, theta, phi);
-    outputDetInfo.setPosition(i, newPos);
+
+    const size_t detIndex = inputSpecInfo.detector(i).index();
```

Comment:
Don't get the detector, that is unnecessary overhad, and `Detector` will eventually be removed. Prefer using `SpectrumInfo::spectrumDefinition`.